core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -71,6 +100,9 @@ private[spark] abstract class LauncherBackend {
     if (connection != null && lastState != state) {
       connection.send(new SetState(state))
       lastState = state
+      if (!_isConnected && stopOnShutdown) {
```

Comment:
why do we need this check at all here vs just relying on the close() call?  _isConnected only becomes false after calling that which should do the fireStopRequest(), am I missing a case?  If it wasn't connected at all on startup the connection != null case wouldn't pass